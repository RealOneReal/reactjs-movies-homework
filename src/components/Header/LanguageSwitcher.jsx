import { FormControl, Select, MenuItem } from "@mui/material";
import { useState } from "react";
import { useDispatch } from "react-redux";
import { language } from "../../redux/search/searchSlice";

const LanguageSwitcher = () => {
  const [languageFromInput, setLanguage] = useState("EN");
  const dispatch = useDispatch();

  const handleChange = ({ target }) => {
    setLanguage(target.value);
    dispatch(language(target.value));
  };

  const styles = {
    menuItem: {
      color: "secondary.dark",
    },
    select: {
      color: "primary.dark",
      "& .MuiSvgIcon-root": {
        color: "primary.dark",
      },
    },
  };

  return (
    <FormControl>
      <Select
        value={languageFromInput}
        onChange={handleChange}
        variant="standard"
        color="secondary"
        sx={styles.select}
      >
        <MenuItem value="EN" sx={styles.menuItem}>
          EN
        </MenuItem>
        <MenuItem value="RU" sx={styles.menuItem}>
          RU
        </MenuItem>
      </Select>
    </FormControl>
  );
};

export default LanguageSwitcher;
