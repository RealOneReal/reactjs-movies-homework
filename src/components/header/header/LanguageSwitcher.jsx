import { FormControl, Select, MenuItem } from "@mui/material";
import { useState } from "react";

const LanguageSwitcher = () => {
  const [language, setLanguage] = useState("EN");

  const handleChange = (event) => {
    setLanguage(event.target.value);
  };

  return (
    <FormControl>
      <Select value={language} onChange={handleChange} variant="standard">
        <MenuItem value="EN">EN</MenuItem>
        <MenuItem value="RU">RU</MenuItem>
      </Select>
    </FormControl>
  );
};

export default LanguageSwitcher;
