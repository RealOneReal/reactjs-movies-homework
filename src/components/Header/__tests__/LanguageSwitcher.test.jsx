import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import LanguageSwitcher from "../LanguageSwitcher";

describe("LanguageSwitcher", () => {
  it("should be render", () => {
    const switcher = renderer.create(<LanguageSwitcher />);
    const tree = switcher.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should have default value EN ", () => {
    const { getByText } = render(<LanguageSwitcher />);
    const value = getByText(/EN/i);
    expect(value).toHaveTextContent("EN");
  });
  it("should have possibility to change value for RU", () => {
    const { getByRole, getByText } = render(<LanguageSwitcher />);
    const button = getByRole("button");
    userEvent.click(button);
    const option = getByText(/RU/i);
    userEvent.click(option);
    const chosenValue = getByRole("button");
    expect(chosenValue).toHaveTextContent("RU");
  });
});
