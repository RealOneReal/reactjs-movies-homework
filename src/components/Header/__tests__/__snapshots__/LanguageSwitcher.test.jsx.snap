// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSwitcher should be render 1`] = `
<div
  className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
>
  <div
    className="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorSecondary MuiInputBase-formControl MuiSelect-root css-h781xk-MuiInputBase-root-MuiInput-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      className="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      EN
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="EN"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
